onlineteamscan_v2_fe/pages/teams.vue

```diff
@@ -1,15 +1,305 @@
+        <v-icon v-if="sorted === 'isTeamscanActive'" dense color="#919191">{{ direction === 1 ? 'mdi-arrow-up' : 'mdi-arrow-down'}}</v-icon>
+      </v-col>
+      <v-col align="center" lg="2"></v-col>
+    </v-row>
+    </div>
+      <v-container v-else class="empty-teams-container">
+          <v-row justify="center">
+            <img src="../static/EmptyIcon.svg" class="empty-teams-img">
+          </v-row>
+          <v-row justify="center">
+            <span class="font-weight-medium header-style">Geen teams gevonden</span>
+          </v-row>
+      </v-container>
+    <div>
+      <custom-list-item class="mx-5 mb-8" v-for="(team, index) in teams" :key="index" :team="team"/>
+    </div>
+
+    <v-dialog v-model="deleteDialog" width="unset">
+      <v-card>
+        <v-card-title>
+          <span class="headline confirmation-card-title">
+            Verwijderen
+          </span>
+        </v-card-title>
+        <v-card-text class="pb-0">
+          Weet u zeker dat u dit team wilt verwijderen?
+        </v-card-text>
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn color="blue darkin-1" text @click="closeDelete">Cancel</v-btn>
+          <v-btn color="blue darkin-1" text @click="confirmDeleteTeam">Ok</v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </div>
 </template>
 
 <script>
+import CustomListItem from "../components/CustomListItem";
+import Button from "../components/Button";
+
 export default {
   components: {
+    CustomListItem,
+    Button,
+  },
+  data() {
+    return {
+      teams: [],
+      originalTeams: [],
+      sorted: '', // Key for sorting the team list
+      direction: 0, // Ascending order of sort
+      isFormValid: false,
+      dialog: false,
+      deleteDialog: false,
+      dialogIndex: -1,
+      originalIndex: -1,
+      errorMessage: '',
+      editedTeam: {
+        name: '',
+        teamleaderId: this.$auth.user.id,
+      },
+      defaultTeam: {
+        name: '',
+        teamleaderId: this.$auth.user.id,
+      },
+      nameRules: [
+        value => !!value || 'Vereist',
+        v => /^[^-\s][ áàíóúéëöüñÄĞİŞȘØøğışÐÝÞðýþa-zA-Z_\s-]*$/.test(v) || 'Moet geldig zijn', /*^[ a-zA-Z\-']+$*/
+        v => v.length <= 50 || 'Max 50 characters'
+      ],
+    }
+  },
+  async created() {
+    await this.$axios.get(`http://localhost:49783/api/teams/teammembers/${this.$auth.user.id}`)
+      .then(response => this.teams = response.data).then(() => this.originalTeams = [...this.teams])
+      .catch(err => console.log(err))
 
+  },
+  watch: {
+    deleteDialog (val) {
+      val || this.closeDelete()
+    },
+    dialog (val) {
+      val || this.close()
+    },
+  },
+  computed: {
+    dialogTitle() {
+      return this.dialogIndex === -1 ? 'Team Toevoegen' : 'Team Bewerken'
+    }
+  },
+  methods: {
+    sortBy(prop) {
+      if (this.sorted === prop && this.direction === -1) {
```

Comment:
probeer zo weinig mogelijk else en elseif te gebruiken.
We kunnen dit enorm leesbaar maken door het zo te optimaliseren.

Spreek me hier morgen even over aan, dan gaan we er over in de middag.